packages/@aws-cdk/aws-appsync/test/appsync-grant.test.ts

```diff
@@ -0,0 +1,1031 @@
+import { join } from 'path';
+import '@aws-cdk/assert/jest';
+import * as iam from '@aws-cdk/aws-iam';
+import * as cdk from '@aws-cdk/core';
+import * as appsync from '../lib';
+
+describe('grant Permissions', () => {
+  test('grant should not throw error', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const role = new iam.Role(stack, 'Role', {
+      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+    });
+    const api = new appsync.GraphQLApi(stack, 'API', {
+      name: 'demo',
+      schemaDefinitionFile: join(__dirname, 'appsync.test.graphql'),
+      authorizationConfig: {
+        defaultAuthorization: {
+          authorizationType: appsync.AuthorizationType.IAM,
+        },
+      },
+    });
+
+    // WHEN
+    const grantResources = [
+      { custom: 'success' },
+    ];
+    const when = () => {
+      api.grant(role, grantResources, 'appsync:graphql');
+    };
+
+    // THEN
+    expect(when).not.toThrow();
```

Comment:
We don't write tests like these. In general, when an exception is thrown, the test will fail. No need to specify that explicitly.

Since this has no assertions, I would just remove it.